kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -69,27 +69,75 @@ function _collectionState(data) {
 
 
 /*
- * Returns a promise that gets current channel.
- */
-function _getCurrentChannel() {
-  let currentChannelId = null;
+* Returns the promise that fetches the channel list.
+*/
+function _getChannelList() {
+  let returnChannelList = null;
   return new Promise((resolve, reject) => {
     ChannelResource.getCollection({}).fetch()
       .then((channelList) => {
         if (channelList.length) {
-          const cookieCurrentChannelId = cookiejs.get('currentChannel');
-          if (channelList.some((channel) => channel.id === cookieCurrentChannelId)) {
-            currentChannelId = cookieCurrentChannelId;
-          } else {
-            currentChannelId = channelList[0].id;
-          }
+          returnChannelList = channelList;
         }
-        resolve(currentChannelId);
+        resolve(returnChannelList);
       });
   });
 }
 
 
+/*
+ * Returns a promise that fetches the current channel id.
+ */
+function _getCurrentChannelId() {
+  let currentChannelId = null;
+  return new Promise((resolve, reject) => {
+    const channelListPromise = _getChannelList();
+    Promise.all([channelListPromise])
```

Comment:
No need to use `Promise.all` here, as you are only waiting on one promise.

Can refactor to:
```
channelListPromise.then((channelList) => {
```